src/annotation.rs

```diff
@@ -6,125 +6,95 @@
 {
-}
-
-impl<K, S> Borrow<Max<K>> for MapAnnotationDefault<K, S>
-where
-    K: Canon<S> + Ord + Default,
-    S: Store,
-{
-    fn borrow(&self) -> &Max<K> {
+    fn borrow(&self) -> &MaxKey<K> {
         &self.max
     }
 }
 
-impl<K, S> Borrow<Cardinality> for MapAnnotationDefault<K, S>
+impl<K> Borrow<Cardinality> for MapAnnotationDefault<K>
 where
-    K: Canon<S> + Ord + Default,
-    S: Store,
+    K: Canon + Ord + Default,
 {
     fn borrow(&self) -> &Cardinality {
         &self.cardinality
     }
 }
 
-impl<K, V, S> Annotation<KelvinMap<K, V, MapAnnotationDefault<K, S>, S>, S>
-    for MapAnnotationDefault<K, S>
+impl<K, V> Annotation<Leaf<K, V>> for MapAnnotationDefault<K>
 where
-    K: Canon<S> + Ord + Default,
-    V: Canon<S>,
-    S: Store,
+    K: Canon + Ord + Default,
 {
-    fn identity() -> Self {
-        let cardinality = <Cardinality as Annotation<
-            KelvinMap<K, V, MapAnnotationDefault<K, S>, S>,
-            S,
-        >>::identity();
-        let max = Max::Maximum(K::default());
-
-        Self {
-            cardinality,
-            max,
-            store: PhantomData,
-        }
-    }
-
     fn from_leaf(leaf: &Leaf<K, V>) -> Self {
-        let cardinality = <Cardinality as Annotation<
-            KelvinMap<K, V, MapAnnotationDefault<K, S>, S>,
-            S,
-        >>::from_leaf(leaf);
-        let max = <Max<K> as Annotation<
-            KelvinMap<K, V, MapAnnotationDefault<K, S>, S>,
-            S,
-        >>::from_leaf(leaf);
+        let cardinality =
+            <Cardinality as Annotation<Leaf<K, V>>>::from_leaf(leaf);
+        let max = <MaxKey<K> as Annotation<Leaf<K, V>>>::from_leaf(leaf);
 
-        Self {
-            cardinality,
-            max,
-            store: PhantomData,
-        }
+        Self { cardinality, max }
     }
+}
 
-    fn from_node(
-        node: &KelvinMap<K, V, MapAnnotationDefault<K, S>, S>,
-    ) -> Self {
-        let cardinality = <Cardinality as Annotation<
-            KelvinMap<K, V, MapAnnotationDefault<K, S>, S>,
-            S,
-        >>::from_node(node);
-        let max = <Max<K> as Annotation<
-            KelvinMap<K, V, MapAnnotationDefault<K, S>, S>,
-            S,
-        >>::from_node(node);
+impl<K, V>
+    Combine<KelvinMap<K, V, MapAnnotationDefault<K>>, MapAnnotationDefault<K>>
+    for MapAnnotationDefault<K>
+where
+    K: Canon + Ord + Default,
+    V: Canon,
+{
+    fn combine(node: &KelvinMap<K, V, MapAnnotationDefault<K>>) -> Self {
+        let cardinality = <Cardinality as Combine<
+            KelvinMap<K, V, MapAnnotationDefault<K>>,
+            MapAnnotationDefault<K>,
+        >>::combine(node);
 
-        Self {
-            cardinality,
-            max,
-            store: PhantomData,
-        }
+        let max = <MaxKey<K> as Combine<
+            KelvinMap<K, V, MapAnnotationDefault<K>>,
+            MapAnnotationDefault<K>,
+        >>::combine(node);
+
+        Self { cardinality, max }
```

Comment:
Same as above. You can reduce the code complexity:
```suggestion
let cardinality = Cardinality::combine(node);
        let max = MaxKey::combine(node);

        Self { cardinality, max }
```

Reply:
Fixed, was using the old required syntax